common/djangoapps/enrollment/handlers.py

```diff
@@ -0,0 +1,67 @@
+"""
+Handlers and actions related to enrollment.
+"""
+import logging
+
+from edxmako.shortcuts import render_to_string
+from django.conf import settings
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from django.core.urlresolvers import reverse
+from django.core.mail import send_mail
+from django.dispatch import receiver
+
+from student.models import ENROLL_STATUS_CHANGE, EnrollStatusChange
+
+log = logging.getLogger(__name__)
+
+
+@receiver(ENROLL_STATUS_CHANGE)
+def send_email_to_staff_on_student_enrollment(sender, event=None, user=None, **kwargs):  # pylint: disable=unused-argument
+    """
+    Sends an e-mail to staff after a new enrollment.
+    This feature can be enabled by setting the e-mail in ENROLLMENT_NOTIFICATION_EMAIL in lms.env.json, or by using
+    a SiteConfiguration variable of the same name. Disabled by default.
+    """
+
+    if event == EnrollStatusChange.enroll:
+        # If there's a SiteConfiguration variable (including ''), we use it instead of the lms.env.json setting
+        to_email = configuration_helpers.get_value('ENROLLMENT_NOTIFICATION_EMAIL',
+                                                   settings.ENROLLMENT_NOTIFICATION_EMAIL)
+
+        if not to_email:
+            # feature disabled
+            return
+
+        course_id = kwargs['course_id']
+
+        site_protocol = 'https' if settings.HTTPS == 'on' else 'http'
+        site_domain = configuration_helpers.get_value('site_domain', settings.SITE_NAME)
+        course_url = u"{protocol}://{domain}{path}".format(
+            protocol=site_protocol,
+            domain=site_domain,
+            path=reverse('about_course', args=[course_id.to_deprecated_string()])
+        )
```

Comment:
Question: I haven't actually looked, but is this the best way to generate these URLs? Wondering if there's any prebuilt function for this as I'd imagine this is a relatively common thing to do.

Reply:
The Django ways to get a URL are in https://stackoverflow.com/questions/2345708/how-can-i-get-the-full-absolute-url-with-domain-in-django, but in this case we don't have a {{request}} object and the offered solutions are also several lines long.
  There might be an openedx way. But I didn't find a {{get_absolute_url}} in CourseLocator, nor a method in CourseMode. The methods that return a path return relative URLs. Grepping for course∩absolute∩url didn't show any code for this.